indigo-plugin/indigo-plugin/src/indigoplugin/generators/AssetListing.scala

```diff
@@ -159,6 +159,21 @@ object AssetListing {
 
             (vals, loadable, named)
 
+          case PathTree.File(name, ext, path) if FontFileExtensions.contains(ext) =>
+            val safeName = toSafeName(name, ext)
+
+            val vals =
+              s"""${indentSpacesNext}val ${safeName}: AssetName               = AssetName("${name}")
+              |${indentSpacesNext}val ${safeName}FontFamily: FontFamily    = FontFamily(${safeName}.toString())"""
```

Comment:
I'm running out of time today - just digging around this in the background as you know.

Superficially this all looks good - and adding the `FontFamily` instance is a good idea! - the catch is that I'm worried there are subtle bugs here due to the flaky nature of font loading. There might not be, I'm just unsure and feeling paranoid.

The cause of the problem you correctly identified is the `.` in the file name, and sure enough, the way we register the font is:

```scala
  def loadFontAsset(fontAsset: AssetType.Font): Future[LoadedFontAsset] =
    IndigoLogger.info(s"[Font] Loading ${fontAsset.path}")

    val font = new FontFace(fontAsset.name.toString, s"url(${fontAsset.path.toString})")

    font.load().toFuture.map { fontFace =>
      IndigoLogger.info(s"[Font] Success ${fontAsset.path}")

      // add font to document
      js.Dynamic.global.document.fonts.add(font)
      // enable font with CSS class
      js.Dynamic.global.document.body.classList.add("indigo-fonts-loaded")

      LoadedFontAsset(AssetName(fontFace.family))
    }
```

So the dot would be used, even though it's invalid. But the `ext` in the asset name was helping keep the assets unique, and I'm not sure the use of `safeName` on the next line will necessarily always line up to the same name you've register the font against.

I need to think about this a little more.

Reply:
Yes, that makes sense. OK, well seeing as it's just the dot that's causing issues, I'll just change the dot to an underscore and all should be fine